x-pack/metricbeat/module/containerd/cpu/cpu.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cpu
+
+import (
+	"github.com/elastic/beats/v7/metricbeat/helper/prometheus"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/metricbeat/mb/parse"
+)
+
+const (
+	defaultScheme = "http"
+	defaultPath   = "/v1/metrics"
+)
+
+var (
+	// HostParser validates Prometheus URLs
+	hostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		DefaultPath:   defaultPath,
+	}.Build()
+)
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	// Mapping of state metrics
+	mapping := &prometheus.MetricsMapping{
```

Comment:
Is there a reason why this map is declared here? I don't think that `getMetricsetFactory` function is getting reused, so we could just put it in one place?